openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,898 @@
+        scaled_points : array
+            Array of normalized slider positions.
+        training_points : array
+            Array of input training data.
+        source : str
+            Which subplot the method is being called from.
+
+        Returns
+        -------
+        idx : array
+            Index of closest points that are within the dist range.
+        x_tree : array
+            One dimentional array of points that are within the dist range.
+        """
+        # Column of the input
+        if source == 'right':
+            col_idx = 1
+        else:
+            col_idx = 0
+
+        # Delete the axis of input from source to predicted 1D distance
+        x = np.delete(scaled_points, col_idx, axis=0)
+        x_training_points = np.delete(training_points, col_idx, axis=1).flatten()
+
+        # Tree of point distances
+        x_tree = np.abs(x - x_training_points)
+
+        # Only return points that are within our distance-viewing paramter.
+        idx = np.where(x_tree <= self.dist_range)
+        x_tree = x_tree[idx]
+        return x_tree, idx[0]
+
+    def _multidimension_input(self, scaled_points, training_points, source='bottom'):
+
+        # Column of the input
+        if source == 'right':
+            col_idx = self.y_input_select.options.index(self.y_input_select.value)
+
+        else:
+            col_idx = self.x_input_select.options.index(self.x_input_select.value)
+
+        # Delete the axis of input from source to predicted distance
+        x = np.delete(scaled_points, col_idx, axis=0)
+        x_training_points = np.delete(training_points, col_idx, axis=1)
+
+        # Tree of point distances
+        x_tree = cKDTree(x_training_points)
+
+        # Query the nearest neighbors tree for the closest points to the scaled array
+        dists, idx = x_tree.query(x, k=len(x_training_points),
+                                  distance_upper_bound=self.dist_range)
+
+        # kdtree query always returns requested k even if there are not enough valid points
+        idx_finite = np.where(np.isfinite(dists))
+        dists_finite = dists[idx_finite]
+        idx = idx[idx_finite]
+        return dists_finite, idx
+
+    # Event handler functions
+    def _update_all_plots(self):
+        self.doc_layout.children[0] = self._contour_data()
+        self.doc_layout.children[1] = self._right_plot()
+        self.doc_layout2.children[0] = self._bottom_plot()
+
+    def _update_subplots(self):
+        self.doc_layout.children[1] = self._right_plot()
+        self.doc_layout2.children[0] = self._bottom_plot()
+
+    def _update(self, attr, old, new):
+        self._update_all_plots()
+
+    def _scatter_plots_update(self, attr, old, new):
+        self._update_subplots()
+
+    def _scatter_input(self, attr, old, new):
+        # Text input update function of dist range value
+        self.dist_range = float(new)
+        self._update_all_plots()
+
+    def _input_dropdown_checks(self, x, y):
```

Comment:
this function doesn't need to exist.  Just replace lines like 'if not self._input_dropdown_checks(new, self.y_input_select.value)` with 'if new == self.y_input_select.value'